src/creator.rs

```diff
@@ -111,24 +111,45 @@ impl<H: Hasher> Creator<H> {
 
     pub(crate) async fn create(&mut self, exit: oneshot::Receiver<()>) {
         self.create_unit();
-        delay_for((self.create_lag)(0)).await;
-        let mut round: usize = 1;
+        let mut round: usize = 0;
         let mut exit = exit.into_stream();
-        loop {
-            tokio::select! {
-                Some(u) = self.parents_rx.next() => {
-                    self.add_unit(u.round(), u.creator(), u.hash());
-                    if self.check_ready() {
-                        self.create_unit();
-                        delay_for((self.create_lag)(round)).await;
-                        round += 1;
+        let mut delay_fut = delay_for(Duration::from_millis(0));
+        'outer: loop {
```

Comment:
Maybe factor out the two inner loops into something like `delay_creation` and `create_asap` (hopefully with better names...), both returning whether they received an exit signal. This might make the structure easier to understand (with good names the comments would hopefully be unnecessary) and would remove the label, to which I have a personal aversion.

Reply:
Got rid of the double loop, and the nested loop, using @mike1729 suggestion. Hopefully this looks better now :)